server/portal/apps/auth/models.py

```diff
@@ -81,7 +83,8 @@ def client(self):
         with transaction.atomic():
             if self.expired:
                 try:
-                    client.refresh_tokens()
+                    with lock:
```

Comment:
Right now this throws an error: `NameError: name 'lock' is not defined`. I believe it might need to be in the context of the transaction, but either way it does work if the `lock` is instantiated right before `with lock` is referenced. Can we move it here? i.e.:
```
                try:
                    lock = Lock()
                    with lock:
                        client.refresh_tokens()
```

Reply:
the self is missing, right?  `with self.lock`

Reply:
Ah yep that's it

Reply:
Great point @chandra-tacc , thanks